make/hotspot/lib/JvmFeatures.gmk

```diff
@@ -33,6 +33,8 @@ ifeq ($(INCLUDE), true)
 # jvm features are selected for this jvm variant.
 ################################################################################
 
+JVM_CFLAGS_FEATURES += -DJVM_VARIANT=\"$(JVM_VARIANT)\"
```

Comment:
Might it be a better idea to set \-DJVM\_VARIANT as the CXXFLAGS for cdsConfig.cpp directly instead?

Reply:
Currently, os_bsd.cpp also needs the JVM_VARIANT definition. Later, if we fix the `os::jvm_path()` issues, os_windows.cpp, os_linux.cpp and os_aix.cpp also need JVM_VARIANT.

Reply:
This is completely wrong.

The JVM_CFLAGS_FEATURES should be used to add feature-specific flags, not like this. If we really did want to add this to the CFLAGS for all Hotspot files, the correct thing to do would be to add it to JVM_CFLAGS in JvmFlags.gmk.

However, that is not something we want to do. I fully agree with Julian. This is a specific patch needed for one file only, and it should be added just for this particular file. Compare how this is done for e.g. `abstract_vm_version.cpp` in CompileJvm.gmk.

Reply:
Maybe we can add a new API `const char* Abstract_VM_Version::vm_variant()` that returns the value defined `-DJVM_VARIANT`? Files like os_*cpp and CDS can call this function to get the string such as "server" or "client".

Reply:
I've pushed another commit based on the above comments.